plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java

```diff
@@ -110,18 +112,38 @@ private T tryConstructor(Constructor<T> constructor, Mapping config, Configurati
 
             for (int i = 0; i < names.length; i++) {
                 final CNode value = config.get(names[i]);
-                if (value == null && parameters[i].getAnnotation(Nonnull.class) != null) {
-                    throw new ConfiguratorException(names[i] + " is required to configure " + target);
-                }
                 final Class t = parameters[i].getType();
+
+                Class<?> clazz = constructor.getDeclaringClass();
+                if (value == null && (parameters[i].isAnnotationPresent(Nonnull.class)   ||
+                    constructor.isAnnotationPresent(ParametersAreNonnullByDefault.class) ||
+                    clazz.isAnnotationPresent(ParametersAreNonnullByDefault.class)       ||
+                    clazz.getPackage().isAnnotationPresent(ParametersAreNonnullByDefault.class))) {
+
+                    if (Set.class.isAssignableFrom(t)) {
+                        LOGGER.log(Level.INFO, "The parameter to be set is @Nonnull but is not present; " +
+                                                           "setting equal to empty set.");
+                        args[i] = Collections.emptySet();
+                    } else if (List.class.isAssignableFrom(t)) {
+                        LOGGER.log(Level.INFO, "The parameter to be set is @Nonnull but is not present; " +
+                                                           "setting equal to empty list.");
+                        args[i] = Collections.emptyList();
+                    } else if (String.class.isAssignableFrom(t)) {
```

Comment:
Will this work? It avoids the issue in the SonarQube plugin

Reply:
@AbhyudayaSharma FYI with our install after the jcasc plugin got it's update our servers are failing to start with:
```
io.jenkins.plugins.casc.ConfiguratorException: triggers is required to configure class hudson.plugins.sonar.SonarInstallation
	at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.tryConstructor(DataBoundConfigurator.java:134)
	at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.instance(DataBoundConfigurator.java:77)
	at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:268)
	at io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator.check(DataBoundConfigurator.java:101)
	at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:345)
	at io.jenkins.plugins.casc.BaseConfigurator.check(BaseConfigurator.java:288)
	at io.jenkins.plugins.casc.BaseConfigurator.configure(BaseConfigurator.java:353)
	at io.jenkins.plugins.casc.BaseConfigurator.check(BaseConfigurator.java:288)
	at io.jenkins.plugins.casc.ConfigurationAsCode.lambda$checkWith$6(ConfigurationAsCode.java:684)
	at io.jenkins.plugins.casc.ConfigurationAsCode.invokeWith(ConfigurationAsCode.java:646)
Caused: io.jenkins.plugins.casc.ConfiguratorException: unclassified: error configuring 'unclassified' with class 
```

Our config snippet looks like:
```
unclassified:
    sonarglobalconfiguration:
        buildWrapperEnabled: true
        installations:
        -   credentialsId: sonarqube_id
            name: our-sonarqube
            serverUrl: https://sonar.ours.url.internal
```

Reply:
Triggers is required, the sonar plugin needs fixing
https://github.com/SonarSource/sonar-scanner-jenkins/blob/master/src/main/java/hudson/plugins/sonar/SonarInstallation.java#L84-L92